parcels/particleset.py

```diff
@@ -122,6 +105,46 @@ def cptr(i):
         else:
             raise ValueError("Latitude and longitude required for generating ParticleSet")
 
+    @classmethod
+    def from_list(cls, grid, pclass, lon, lat):
+        """Initialise the ParticleSet from start/finish coordinates
+        with equidistant spacing
+
+        :param grid: :mod:`parcels.grid.Grid` object from which to sample velocity
+        :param pclass: mod:`parcels.particle.JITParticle` or :mod:`parcels.particle.ScipyParticle`
+                 object that defines custom particle
+        """
+        return cls(grid=grid, pclass=pclass, lon=lon, lat=lat)
+
+    @classmethod
+    def from_line(cls, grid, pclass, start, finish, size):
+        """Initialise the ParticleSet from start/finish coordinates
+        with equidistant spacing
+
+        :param grid: :mod:`parcels.grid.Grid` object from which to sample velocity
+        :param pclass: mod:`parcels.particle.JITParticle` or :mod:`parcels.particle.ScipyParticle`
+                 object that defines custom particle
+        :param start: Starting point for initialisation of particles on a straight line.
+        :param finish: End point for initialisation of particles on a straight line.
+        :param size: Initial size of particle set
+        """
+        lon = np.linspace(start[0], finish[0], size, dtype=np.float32)
+        lat = np.linspace(start[1], finish[1], size, dtype=np.float32)
```

Comment:
I think that providing for geodetic deformations (which I assume you're referring to; i.e. when a shortest line between two points on a globe is not a straight line) is actually quite complicated to compute.

My philosophy would be to only provide 'simple' functions with parcels, and have users code up (and share?) their more complicated ones themselves. Since you can always initiate a particleset from a list of lons and lats, users can create this list using as complicated functions as they want...